src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/NewAzureContainerRegistryWebhook.cs

```diff
@@ -0,0 +1,125 @@
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.New, ContainerRegistryWebhookNoun, DefaultParameterSetName = NameResourceGroupParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSContainerRegistryWebhook))]
+    public class NewAzureContainerRegistryWebhook : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "Webhook Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(WebhookNameAlias, ResourceNameAlias)]
+        public string Name { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string RegistryName { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, HelpMessage = "The service URI for the webhook to post notifications.")]
+        [Alias(WebhookUriAlias)]
+        public Uri Uri { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, HelpMessage = "Space separated list of actions that trigger the webhook to post notifications.")]
+        [Alias(WebhookActionsAlias)]
+        [ValidateSet(WebhookAction.Delete, WebhookAction.Push)]
+        public string[] Action { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RegistryObjectParameterSet, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
+        public PSContainerRegistry Registry { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Custom headers that will be added to the webhook notifications.")]
+        [ValidateNotNull]
+        [Alias(WebhookHeadersAlias)]
+        public Hashtable Header { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Webhook tags.")]
+        [ValidateNotNull]
+        [Alias(TagsAlias)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Webhook status")]
+        [Alias(WebhookStatusAlias)]
+        [ValidateSet(WebhookStatus.Enabled, WebhookStatus.Disabled)]
+        public string Status { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Webhook scope.")]
+        [ValidateNotNull]
+        [Alias(WebhookScopeAlias)]
+        public string Scope { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Webhook Location. Default to the location of the registry.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(WebhookLocationAlias)]
+        public string Location { get; set; }
```

Comment:
@Jinhuafei for your creation cmdlets, consider adding the `LocationCompleter` attribute to your `Location` parameters: https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#location-completer

This will allow users to tab through the valid locations in which they can create the resource (based on the resource type provided in the completer attribute).

Reply:
fixed